sbi/inference/snle/snle_base.py

```diff
@@ -295,3 +247,82 @@ def _train(
         # Update summary.
         self._summary["epochs"].append(epoch)
         self._summary["best_validation_log_probs"].append(self._best_val_log_prob)
+
+        # Update TensorBoard and summary dict.
+        self._summarize(
+            round_=self._round, x_o=None, theta_bank=theta, x_bank=x,
+        )
+
+        # Update description for progress bar.
+        if show_train_summary:
+            print(self._describe_round(self._round, self._summary))
+
+        return deepcopy(self._neural_net)
+
+    def build_posterior(
+        self,
+        density_estimator: Optional[TorchModule] = None,
```

Comment:
why do we need this? this is a bit confusing I find.

Reply:
I added it because `.train()` returns the `density_estimator`. So users might be tempted to do:
```python
density_estimator = inference.train()
posterior = inference.build_posterior(density_estimator)
```

When it is not passed, we just use the latest trained network. Thus, the above is equivalent to:
```python
density_estimator = inference.train()
posterior = inference.build_posterior()
```

I'm ok with deleting the argument if you think it is clearer.

Reply:
I see, you are right, it makes sense to have the `density_estimator` arg.

Reply:
Now I get this as well 👍🏽